package.json

```diff
@@ -74,6 +74,19 @@
     "eslint-plugin-import": "^2.27.5",
     "eslint-plugin-jsx-a11y": "^6.7.1",
     "eslint-plugin-react": "^7.32.2",
-    "eslint-plugin-react-hooks": "^4.6.0"
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "husky": "^9.0.10",
+    "lint-staged": "^15.2.2"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "src/**/*.{js,jsx,ts,tsx}": [
+      "prettier --write",
+      "eslint --fix"
+    ]
```

Comment:
I appreciate your work on this. Could we look into configuring the script to run during pull requests instead of commits? This way, we can avoid adding extra dependencies and save time during development by ensuring it runs only when necessary. Perhaps we could start by keeping things simple for now and focus on maintaining the script within our existing setup by keeping `Line 16` and `Line 17` in .`eslintrc.json`.

Reply:
I think it should be a separate ticket(issue) created for further work. I adjusted the code according to the PR comment